source/math/FloatPoint2DVectorList.ooc

```diff
@@ -87,6 +87,54 @@ FloatPoint2DVectorList: class extends VectorList<FloatPoint2D> {
 		}
 		result
 	}
+	offset: func (offset: FloatPoint2D) -> This {
+		result := This new()
+		thisPointer := (this pointer as FloatPoint2D*)
+		for (i in 0 .. this count)
+			result add(thisPointer[i] + offset)
+		result
+	}
+	_swap: static func (array: FloatPoint2D*, i, j: Int) {
+		t := array[i]
+		array[i] = array[j]
+		array[j] = t
+	}
+	_partition: static func (array: FloatPoint2D*, start, end, pivot: Int) -> Int {
+		pivotValue := array[pivot] x
+		This _swap(array, pivot, end)
+		result := start
+		for (i in start .. end)
+			if (array[i] x < pivotValue) {
+				This _swap(array, result, i)
+				++result
+			}
+		This _swap(array, result, end)
+		result
+	}
+	_medianOfThree: static func (array: FloatPoint2D*, start, end: Int) -> Int {
+		mid := (start + end) / 2
+		if (array[start] x > array[mid] x)
+			This _swap(array, mid, start)
+		if (array[mid] x > array[end] x)
+			This _swap(array, mid, end)
+		if (array[start] x > array[end] x)
+			This _swap(array, start, end)
+		mid
+	}
+	_quicksort: static func (array: FloatPoint2D*, start, end: Int) {
+		if (end == start + 1 && array[start] x > array[end] x)
+			This _swap(array, start, end)
+		else if (start < end) {
+			pivot := This _partition(array, start, end, This _medianOfThree(array, start, end))
+			if (pivot > start)
+				This _quicksort(array, start, pivot - 1)
+			if (pivot < end)
+				This _quicksort(array, pivot + 1, end)
+		}
+	}
+	sortByX: func {
+		This _quicksort(this _vector _backend as FloatPoint2D*, 0, this count - 1)
```

Comment:
I'm not sure we want to add the sort in this way. I basically just copied the code from FloatVectorList and modified it to use FloatPoint2D. I didn't add a sortByY because I think sortByX would be the most common usage.